test/common/PrepDataFuncs.cpp

```diff
@@ -0,0 +1,342 @@
+/*************************************************************************
+ * Copyright (c) 2022 Advanced Micro Devices, Inc. All rights reserved.
+ *
+ * See LICENSE.txt for license information
+ ************************************************************************/
+
+#include "CollectiveArgs.hpp"
+#include "PrepDataFuncs.hpp"
+#include <cstdio>
+#include <hip/hip_runtime.h>
+
+namespace RcclUnitTesting
+{
+  ErrCode DefaultPrepareDataFunc(CollectiveArgs &collArgs)
+  {
+    switch (collArgs.funcType)
+    {
+    case ncclCollBroadcast:     return DefaultPrepData_Broadcast(collArgs);
+    case ncclCollReduce:        return DefaultPrepData_Reduce(collArgs, false);
+    case ncclCollAllGather:     return DefaultPrepData_Gather(collArgs, true);
+    case ncclCollReduceScatter: return DefaultPrepData_ReduceScatter(collArgs);
+    case ncclCollAllReduce:     return DefaultPrepData_Reduce(collArgs, true);
+    case ncclCollGather:        return DefaultPrepData_Gather(collArgs, false);
+    case ncclCollScatter:       return DefaultPrepData_Scatter(collArgs);
+    case ncclCollAllToAll:      return DefaultPrepData_AllToAll(collArgs);
+      //case ncclCollSendRecv:      return DefaultPrepData_SendRecv(collArgs);
+    default:
+      ERROR("Unknown func type %d\n", collArgs.funcType);
+      return TEST_FAIL;
+    }
+  }
+
+  ErrCode CheckAllocation(CollectiveArgs const& collArgs)
+  {
+    if (collArgs.numInputElements > collArgs.numInputElementsAllocated)
+    {
+      ERROR("Number of input elements (%lu) exceeds the number of allocated input elements (%lu)\n",
+            collArgs.numInputElements, collArgs.numInputElementsAllocated);
+      return TEST_FAIL;
+    }
+
+    if (collArgs.numOutputElements > collArgs.numOutputElementsAllocated)
+    {
+      ERROR("Number of output elements (%lu) exceeds the number of allocated output elements (%lu)\n",
+            collArgs.numOutputElements, collArgs.numOutputElementsAllocated);
+      return TEST_FAIL;
+    }
+    return TEST_SUCCESS;
+  }
+
+  ErrCode DefaultPrepData_Broadcast(CollectiveArgs &collArgs)
+  {
+    if (CheckAllocation(collArgs) != TEST_SUCCESS) return TEST_FAIL;
```

Comment:
Simple `CHECK_CALL(CheckAllocation(collArgs))` should work here instead right? I assume that the extra line: `ERROR("Error in call %s\n", #func);  ` in check call wont be a problem in this case